lib/index.js

```diff
@@ -52,6 +57,14 @@ function getText(fieldBoost, doc) {
   return text;
 }
 
+function handleNestedObjectArrayItem(fieldBoost, deepField) {
+  return function (one) {
+    return getText(utils.extend({}, fieldBoost, {
+      deepField: deepField
+    }), one);
+  };
```

Comment:
is it necessary to make this a function that returns a function? typically better for perf to just declare one function once